v1/src/main/java/com/google/cloud/teleport/templates/common/DatadogConverters.java

```diff
@@ -0,0 +1,393 @@
+                            // See:
+                            // https://cloud.google.com/logging/docs/reference/v2/rest/v2/MonitoredResource
+                            JSONObject resource = data.optJSONObject(GCP_RESOURCE_KEY);
+                            boolean resourceAvailable = (resource != null && !resource.isEmpty());
+
+                            if (resourceAvailable) {
+
+                              // Check if the resource object has a type string
+                              // If so, convert it to a Datadog source string and add it to the
+                              // DatadogEvent, i.e.
+                              // "gce_instance"
+                              // converts to
+                              // "gcp.gce.instance"
+                              String type = resource.optString(GCP_RESOURCE_TYPE_KEY);
+                              if (!type.isEmpty()) {
+                                String formattedSource =
+                                    DD_DEFAULT_SOURCE + "." + type.replaceAll("_", ".");
+                                builder.withSource(formattedSource);
+                              }
+
+                              // Check if the resource object has a labels object
+                              // If so, convert it to a Datadog tags string and add it to the
+                              // DatadogEvent, i.e.
+                              // {"projectId": "my-project", "instanceId": "12345678901234", "zone":
+                              // "us-central1-a"}
+                              // converts to
+                              // "projectId:my-project,instanceId:12345678901234,zone:us-central1-a"
+                              JSONObject labels = resource.optJSONObject(GCP_RESOURCE_LABELS_KEY);
+                              boolean labelsAvailable = (labels != null && !labels.isEmpty());
+
+                              if (labelsAvailable) {
+                                List<String> tags = new ArrayList<>();
+                                for (Map.Entry<String, Object> label : labels.toMap().entrySet()) {
+                                  String labelName = label.getKey();
+                                  String labelValue = label.getValue().toString();
+                                  if (labelName.isEmpty() || labelValue.isEmpty()) {
+                                    continue;
+                                  }
+
+                                  tags.add(String.format("%s:%s", labelName, labelValue));
+                                }
+
+                                String formattedTags = Joiner.on(",").join(tags);
+                                if (!formattedTags.isEmpty()) {
+                                  builder.withTags(formattedTags);
+                                }
+                              }
+                            }
+                          }
```

Comment:
Right now, this kind of processing occurs on the Datadog side as part of Logs Intake, specifically when `protocol=gcp` is set, as [our documentation requires](https://docs.datadoghq.com/integrations/google_cloud_platform/#forward-logs-to-datadog).
- turn `$.resource.type` into `ddsource`
- turn `$.resource.labels` into `ddtags`